patches/chromium/convert_electron_processsingleton_changes_to_patch.patch

```diff
@@ -0,0 +1,159 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Raymond Zhao <raymondzhao@microsoft.com>
+Date: Wed, 18 Aug 2021 08:24:10 -0700
+Subject: Convert Electron ProcessSingleton changes to patch
+
+This patch applies Electron ProcessSingleton changes
+onto the Chromium files, so that Electron doesn't need to maintain
+separate copies of those files.
+
+This patch adds a few changes to the Chromium code:
+1. It adds a parameter `program_name` to the constructor so that
+   the generated mutex name on the Windows-side is program-dependent,
+   rather than shared between all Electron applications.
+2. It adds an `IsAppSandboxed` check to the posix side so that
+   sandboxed posix applications generate shorter temp paths.
+3. It adds a `ChangeWindowMessageFilterEx` call to the Windows
+   implementation to handle the case when the primary app is run with
+   admin permissions.
+
+diff --git a/chrome/browser/process_singleton.h b/chrome/browser/process_singleton.h
+index 0291db0f919aa7868d345f4f0712c42c6ad7ee17..1c34b939e85543aba6801c45ba81fe626aad65fe 100644
+--- a/chrome/browser/process_singleton.h
++++ b/chrome/browser/process_singleton.h
+@@ -103,7 +103,8 @@ class ProcessSingleton {
+       base::RepeatingCallback<bool(const base::CommandLine& command_line,
+                                    const base::FilePath& current_directory)>;
+ 
+-  ProcessSingleton(const base::FilePath& user_data_dir,
++  ProcessSingleton(const std::string& program_name,
++                   const base::FilePath& user_data_dir,
+                    const NotificationCallback& notification_callback);
+   ~ProcessSingleton();
+ 
+@@ -166,6 +167,7 @@ class ProcessSingleton {
+ 
+  private:
+   NotificationCallback notification_callback_;  // Handler for notifications.
++  std::string program_name_; // Used for kMutexName.
```

Comment:
polish note to self: this should be `mutexName` as `kMutexName` isn't a variable now.